packages/tests/tests/utils.spec.ts

```diff
@@ -266,3 +270,37 @@ describe("selectPeerForProtocol", () => {
     );
   });
 });
+
+describe("getConnectedPeersForProtocol", function () {
+  let waku: LightNode;
+  let nwaku: NimGoNode;
+
+  beforeEach(async function () {
+    this.timeout(15000);
+    nwaku = new NimGoNode(makeLogFileName(this));
+    await nwaku.start({
+      filter: true,
+      lightpush: true,
+      relay: true
+    });
+    waku = await createLightNode();
+    await waku.start();
+    await waku.dial(await nwaku.getMultiaddrWithId());
+    await waitForRemotePeer(waku, [Protocols.Filter]);
+  });
+
+  afterEach(async function () {
+    this.timeout(10000);
+    await tearDownNodes(nwaku, waku);
+  });
+
+  it("returns all connected peers that support the protocol", async function () {
```

Comment:
Is there an easy way to have a discovered peer that's disconnected? Might be nice to have a test to ensure that peer isn't included

Reply:
hum can you clarify what you mean? 
like for a peer that doesn't support the protocol but is `discovered` -- it should not be `connected`, right?

--
merging this for now, please feel free or let me know to add a test case for that as a followup